inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/deformable_convolution_gpu_bfyx_ref.cl

```diff
@@ -48,53 +51,74 @@ KERNEL(deformable_convolution_gpu_bfyx_ref)(
+                        int right_x_index  = left_x_index + 1;
+
+                        INPUT0_TYPE top_left     = top_y_index < 0 || left_x_index < 0 ? 0 : (INPUT0_TYPE)data[GET_DATA_INDEX(INPUT0, b, ifm, top_y_index, left_x_index) + in_split_offset];
+                        INPUT0_TYPE top_right    = top_y_index < 0 || right_x_index >= INPUT0_SIZE_X ? 0 :
+                            (INPUT0_TYPE)data[GET_DATA_INDEX(INPUT0, b, ifm, top_y_index, right_x_index) + in_split_offset];
+                        INPUT0_TYPE bottom_left  = bottom_y_index >= INPUT0_SIZE_Y || left_x_index < 0 ? 0 :
+                            (INPUT0_TYPE)data[GET_DATA_INDEX(INPUT0, b, ifm, bottom_y_index, left_x_index) + in_split_offset];
+                        INPUT0_TYPE bottom_right = bottom_y_index >= INPUT0_SIZE_Y || right_x_index >= INPUT0_SIZE_X ? 0 :
+                                (INPUT0_TYPE)data[GET_DATA_INDEX(INPUT0, b, ifm, bottom_y_index, right_x_index) + in_split_offset];
+#else
+                        int bottom_y_index = (int)(min(ceil(transformed_y), (float)INPUT0_SIZE_Y - 1));
+                        int right_x_index  = (int)(min(ceil(transformed_x), (float)INPUT0_SIZE_X - 1));
 
-                            INPUT0_TYPE top    = top_left + (top_right - top_left) * (transformed_x - left_x_index);
-                            INPUT0_TYPE bottom = bottom_left + (bottom_right - bottom_left) * (transformed_x - left_x_index);
+                        INPUT0_TYPE top_left     = (INPUT0_TYPE)data[GET_DATA_INDEX(INPUT0, b, ifm, top_y_index, left_x_index) + in_split_offset];
+                        INPUT0_TYPE top_right    = (INPUT0_TYPE)data[GET_DATA_INDEX(INPUT0, b, ifm, top_y_index, right_x_index) + in_split_offset];
+                        INPUT0_TYPE bottom_left  = (INPUT0_TYPE)data[GET_DATA_INDEX(INPUT0, b, ifm, bottom_y_index, left_x_index) + in_split_offset];
+                        INPUT0_TYPE bottom_right = (INPUT0_TYPE)data[GET_DATA_INDEX(INPUT0, b, ifm, bottom_y_index, right_x_index) + in_split_offset];
+#endif
+                        INPUT0_TYPE top    = top_left + (top_right - top_left) * (transformed_x - left_x_index);
+                        INPUT0_TYPE bottom = bottom_left + (bottom_right - bottom_left) * (transformed_x - left_x_index);
 
-                            INPUT0_TYPE value  = top + (bottom - top) * (transformed_y - top_y_index);
+                        INPUT0_TYPE value  = top + (bottom - top) * (transformed_y - top_y_index);
 
-                            dotProd += value * weights[filter_idx];
-                        }
-                    } // zero_x
-                } // zero_y
+#if DEFORMABLE_MASK_ENABLED
+                        dotProd += value * weights[filter_idx] * (float)mask[mask_idx];
+#else
+                        dotProd += value * weights[filter_idx];
+#endif
+                    }
+                } // !y_is_out_of_boundaries && !x_is_out_of_boundaries
```

Comment:
I suppose it makes sence to remove empty 123 line.

Reply:
done